test/bioprinttest.c

```diff
@@ -11,8 +11,12 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <openssl/opensslconf.h>
-#include OPENSSL_UNISTD
+#include <openssl/e_os2.h>
+#ifdef OPENSSL_SYS_WINDOWS
+# include <winsock.h>
```

Comment:
> So if we have configured no-stdio, we don't run bioprinttest at all?

There is certain ambiguity in terminology. When we refer to BIO_printf we actually rather refer to its core part, more specifically its ability to produce formatted strings. So that when we say BIO_printf we probably rather mean _dopr. But _dopr actually can be tested with BIO_snprintf and memcmp, isn't it? And that's what bioprinttest.c does in its core. So that if things are so bad, that you don't have stdio and exit code is the only way to convey information, you still can perform meaningful test. It's meaningful, because if things are that bad that there is no stdio, you ought to have some lower-level debugging facility, e.g. through JTAG, and should be able to debug in case failure is returned.